src/core/thread/mesh_forwarder.cpp

```diff
@@ -1254,6 +1259,13 @@ ThreadError MeshForwarder::SendFragment(Message &aMessage, Mac::Frame &aFrame)
 
         if (payloadLength > fragmentLength)
         {
+            if (!aMessage.IsLinkSecurityEnabled())
```

Comment:
I think this works, but we should limit this functionality to MLE messages.  I think we can do this by defining a new `Message::mSubType` value, setting it in `Mle::NewMleMessage()`, then checking the sub type value using `aMessage.GetSubType()`.  Thoughts?